### PR TITLE
Add rabbitmq support, and some restructuring

### PR DESCRIPTION
**Pull request by eamonford**

### CODE REVIEW

File: bin/run_collections [hunk 1]
```diff
@@ -29,38 +29,37 @@ parser.add_argument("-fp", "--forward-processing",
...
+                                                                                          "connection_config")),
+                                                     profiles=config.get("INGEST", "connection_profile").split(','),
+                                                     namespace=config.get("INGEST", "kubernetes_namespace"),
+                                                     dry_run=config.getboolean("OPTIONS", "dry_run"))
 
 
 if config.has_option("COLLECTIONS_YAML_CONFIG", "yaml_file"):
-    collections_ingester.read_yaml_collection_config(full_path(config.get('COLLECTIONS_YAML_CONFIG', 'yaml_file')),
-                                                     collection_row_callback)
+    ingestion_order_executor.read_yaml_collection_config(full_path(config.get('COLLECTIONS_YAML_CONFIG', 'yaml_file')),
```
Comment (tloubrieu-jpl) [inline bin/run_collections]: I am not finding this method anymore. I guessed you wanted to replace that with the object FileIngestionOrderStore but   I am not sure...

File: sdap_ingest_manager/__init__.py [hunk 2]
```diff
@@ -1 +0,0 @@
-__version__ = "0.4.0+dev"
```
Comment (tloubrieu-jpl) [inline sdap_ingest_manager/__init__.py:1]: I am using that to document the current version (as describe here https://www.python.org/dev/peps/pep-0396/). There is a continuous integration thing configured with github actions which uses that (.github/workflows/).
That is very comfortable to use.

File: containers/kubernetes/data-volume.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
   persistentVolumeReclaimPolicy: Delete
   storageClassName: hostpath
   hostPath:
-    path: /Users/loubrieu/PycharmProjects/sdap_ingest_manager/sdap_ingest_manager/collections_ingester/test/data
```
Comment (tloubrieu-jpl) [inline containers/kubernetes/data-volume.yml:16]: I've seen that you move all unit test to tests directory at root of the project. That works for me if it is more consistent with what we have on sdap.

File: sdap_ingest_manager/history_manager/DatasetIngestionHistory.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import hashlib
+import logging
+import os
+from abc import ABC, abstractmethod
```
Comment (tloubrieu-jpl) [inline sdap_ingest_manager/history_manager/DatasetIngestionHistory.py:4]: That is interesting and good. I did not know about the abstract class in python

File: sdap_ingest_manager/granule_ingester/runjobs.py [hunk 5]
```diff
@@ -4,20 +4,21 @@
 import datetime
```
Comment (tloubrieu-jpl) [inline sdap_ingest_manager/granule_ingester/runjobs.py]: Do we want to keep this file ? I would say no but may be we need to keep something like that to run the ingestion outside kubernetes ?

File: sdap_ingest_manager/config/LocalConfiguration.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import configparser
+import logging
```
Comment (tloubrieu-jpl) [inline sdap_ingest_manager/config/LocalConfiguration.py:2]: I think I made a simplification on that while working on the git synchronization with the assumption that we will not use the .ini files.
We can discuss that.

File: sdap_ingest_manager/ingestion_order_store/FileIngestionOrderStore.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+from sdap_ingest_manager.ingestion_order_store import IngestionOrderStore
```
Comment (tloubrieu-jpl) [inline sdap_ingest_manager/ingestion_order_store/FileIngestionOrderStore.py:1]: We can take some time tomorrow morning to discuss that. I am not getting how you foresee this IngestionOrderStores to work.

File: setup.py [hunk 8]
```diff
@@ -51,18 +51,18 @@ def post_install_message():
     python_requires='>=3.6',
```
Comment (tloubrieu-jpl) [inline setup.py:53]: python setup.py test  or pytest -s do not work anymore

File: tests/ingestion_order_executor/test_nfs_mount_parse.py [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import unittest
-from sdap_ingest_manager.collections_ingester import nfs_mount_parse
```
Comment (tloubrieu-jpl) [inline tests/ingestion_order_executor/test_nfs_mount_parse.py:2]: I need to think if we still need that. This was meant to enable to "deconstruct" file path seen from a nfs mount point to there server side path. If the ingestion is fully run in kubernetes we don't need that

File: tests/ingestion_order_executor/test_validation.py [hunk 10]
```diff
@@ -1,23 +1,22 @@
-import unittest
```
Comment (tloubrieu-jpl) [inline tests/ingestion_order_executor/test_validation.py]: this was meant to do somekind of integration test. I don't think it is still useful as it is written (ie for the command line entry points)

**Review by tloubrieu-jpl — COMMENTED** [review]
Before merging I would like to make the unit test work (python setup.py test) and document and be able to deploy locally and launch the main job (service.py) in kubernetes.

File: containers/kubernetes/job.yml [hunk 11]
```diff
@@ -22,7 +22,7 @@ spec:
             name: collection-ingester-config
```
Comment (tloubrieu-jpl) [inline containers/kubernetes/job.yml:22]: This file should call service.py is it ?

File: sdap_ingest_manager/granule_ingester/runjobs.py (hunk 5 above)
Comment (eamonford) [reply sdap_ingest_manager/granule_ingester/runjobs.py]: Deleted this module now

File: sdap_ingest_manager/history_manager/DatasetIngestionHistory.py (hunk 4 above)
Comment (eamonford) [reply sdap_ingest_manager/history_manager/DatasetIngestionHistory.py:4]: ABC and abstract methods are python's way of creating abstract base classes

**Pull request merged**